src/metafile.c

```diff
@@ -494,20 +541,29 @@ gdip_metafile_CreatePenIndirect (MetafilePlayContext *context, DWORD style, DWOR
 	}
 	GdipSetPenStartCap (pen, line_cap);
 	GdipSetPenEndCap (pen, line_cap);
+	
+	s = (dwPenStyle & PS_JOIN_MASK);
+	switch (s) {
+	default:
+		g_warning ("Invalid pen join, style %d, (style & PS_JOIN_MASK) %d", dwPenStyle, s);
+		/* fall through */
+	case PS_JOIN_ROUND:
+		line_join = LineJoinRound;
+		break;
+	case PS_JOIN_BEVEL:
+		line_join = LineJoinBevel;
+		break;
+	case PS_JOIN_MITER:
+		line_join = LineJoinMiter;
+		break;
+	}
+	GdipSetPenLineJoin (pen, line_join);
```

Comment:
I still thing this block should be wrapped in `if (dwPenStyle & PS_GEOMETRIC)`.

Reply:
...including the line caps (https://msdn.microsoft.com/en-us/library/windows/desktop/dd162711(v=vs.85).aspx).